lms/djangoapps/certificates/views/webview.py

```diff
@@ -523,30 +522,30 @@ def render_html_view(request, user_id, course_id):
     _update_context_with_basic_info(context, course_id, platform_name, configuration)
     invalid_template_path = 'certificates/invalid.html'
 
-    # Kick the user back to the "Invalid" screen if the feature is disabled
```

Comment:
Why was this logic moved below?  If HTML certificates aren't enabled, why not check it first and eliminate the call to the modulestore?

Reply:
`has_html_certificates_enabled()` used to take a course_key and optionally, a course object, as parameters, fetching the course by id if one wasn't passed in, but returned early if the `CERTIFICATES_HTML_VIEW` setting was off (before loading the course).  So we can do a redundant early check of the HTML certs setting, then load the course, then check if HTML certs are enabled for the course.